src/transformers/tokenization_t5.py

```diff
@@ -57,6 +57,13 @@
     "t5-3b": 512,
     "t5-11b": 512,
 }
+####################################################
+# Mapping from task to prefix pattern that input_str should start with
+####################################################
+T5_PREFIX_PATTERNS = {
+    "summarization": "summarize: ",
+    "translation": "translate {} to {} ",
+}
```

Comment:
If we have task-specific generation parameters (like detailed above), these could find their ways in the task-specific dict of generation parameters (probably cleaner to have them on S3 rather than hard-coded in the library).

Maybe as a `prefix` argument for the `generation` method?

Reply:
I think in the long run this should be part of the T5 Tokenizer, so I am happy to add them to the T5 config for now. 

I don't really like the idea of adding them to the `generation` method as an argument because 
- For me there are part of the input_ids and it is not `generate` method to concatenate prefix and "main" `input_ids` I feel like
- The user would have to insert `tokenizer.encode("summarize: ")` to the `generation` method since it is separated from the tokenizers and there a lot of errors could be created when special tokens are added, batched input is used etc.

We already have a `add_prefix_space` argument for tokenizers, which we could maybe change to add `add_prefix` and then have a `prefix` argument in the config (which would default to '')? @mfuntowicz @n1t0